internal/agent/list.go

```diff
@@ -0,0 +1,94 @@
+package agent
+
+import (
+	"fmt"
+
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/charmbracelet/bubbles/list"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+var agentListStyle = lipgloss.NewStyle().Margin(1, 2)
+
+type agentListItem struct {
+	title, metadata string
+}
+
+func (ali agentListItem) Title() string       { return ali.title }
+func (ali agentListItem) Description() string { return ali.metadata }
+func (ali agentListItem) FilterValue() string { return ali.title }
+
+type agentListModel struct {
+	agentList list.Model
+	agents    []buildkite.Agent
+	quitting  bool
+}
+
+func ObtainAgents(f *factory.Factory, args ...string) (*agentListModel, error) {
+	var alo buildkite.AgentListOptions
+	var items []list.Item
+
+	if args[0] != "" || args[1] != "" || args[2] != "" {
```

Comment:
This line has a high chance to get runtime error if ObtainAgent is called with args < 3

Reply:
As above - though happy to roll to specified, typed args 🙂

Reply:
I switched these to non-variadic args